app/models/invoice.rb

```diff
@@ -0,0 +1,7 @@
+class Invoice < ApplicationRecord
+  belongs_to :customer
+  has_many :invoice_items
+  has_many :items, through: :invoice_items
+
+  validates :status, presence: true, numericality: true
```

Comment:
instead of status i would've used this line here here:
```ruby
enum status: {cancelled: "cancelled", in_progress: "in progress", completed: "completed"}
```

and inside of the db/migrate/invoice_migration I would've used this:
```ruby
create_enum(:enum_status, %w[cancelled in_progress completed])

t.enum(:status, enum_name: :enum_status)
```

Reply:
I like this. By chance, have you found an article or a doc reference that we could read to learn more on how to do this. I was reading last night a little. Also, how does writing the invoice migration like that tie the status to a particular integer... Like "cancelled" == '1'

Reply:
https://naturaily.com/blog/ruby-on-rails-enum 
^This is a really good read up on how enums work!

 I just finished reading this and it clarified a ton about how enums work and best practices on how to use them.